R/prepInputs.R

```diff
@@ -813,13 +808,66 @@ appendChecksumsTable <- function(checkSumFilePath, filesToChecksum,
   filesInArchive <- NULL
   if (!is.null(funWArgs$fun)) {
     if (file.exists(archive[1])) {
-      filesInArchive <- funWArgs$fun(archive[1], list = TRUE)
-      if ("Name" %in% names(filesInArchive)) {
-        # for zips, rm directories (length = 0)
-        filesInArchive <-
-          filesInArchive[filesInArchive$Length != 0,]$Name
-      } else { # untar
-        filesInArchive
+      if (!file_ext(archive[1]) == "rar") {
+        filesInArchive <- funWArgs$fun(archive[1], list = TRUE)
+        if ("Name" %in% names(filesInArchive)) {
+          # for zips, rm directories (length = 0)
+          filesInArchive <-
+            filesInArchive[filesInArchive$Length != 0,]$Name
+        } else { # untar
+          filesInArchive
+        }
+      } else {
+        message(paste0("The archive is a .rar file. preProcess will try a system call of 'unrar'."))
+        wd <- getwd()
+        tempDir <- file.path(dirname(archive[1]), "extractedFiles")
+        dir.create(tempDir, showWarnings = FALSE)
+        setwd(tempDir)
```

Comment:
need to use `on.exit(setwd(wd), add = TRUE)` right away

Reply:
right after `wd <- getwd()`?

Reply:
no, right after `setwd(tempDir)`

Reply:
instead of `dir.create` etc. you can also use:

```r
tempDir <- file.path(dirname(archive[1]), "extractedFiles") %>%
  checkPath(create = TRUE)
```

I prefer this approach because it works more correctly across OSes